service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -94,13 +101,42 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = getCaseData(callbackRequest.getCaseDetails());
         CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
 
-        if (isRemovingNewSDO(caseData, caseDataBefore)) {
+        Optional<StandardDirectionOrder> removedSDO = getRemovedSDO(caseData, caseDataBefore);
+        Optional<CaseManagementOrder> removedCMO = getRemovedCMO(caseData, caseDataBefore);
+
+        if (removedSDO.isPresent()) {
             publishEvent(new PopulateStandardDirectionsEvent(callbackRequest));
+            publishEvent(new StandardDirectionsOrderRemovedEvent(
+                caseData, removedSDO.map(StandardDirectionOrder::getRemovalReason).orElse(null)));
+        } else if (removedCMO.isPresent()) {
+            publishEvent(
+                new CMORemovedEvent(caseData, removedCMO.map(CaseManagementOrder::getRemovalReason).orElse(null)));
         }
     }
 
-    private boolean isRemovingNewSDO(CaseData caseData, CaseData caseDataBefore) {
-        return !Objects.equals(caseData.getHiddenStandardDirectionOrders(),
-            caseDataBefore.getHiddenStandardDirectionOrders());
+    private Optional<StandardDirectionOrder> getRemovedSDO(CaseData caseData, CaseData caseDataBefore) {
+        List<Element<StandardDirectionOrder>> hiddenSDOs = caseData.getHiddenStandardDirectionOrders();
+        List<Element<StandardDirectionOrder>> previousHiddenSDOs = caseDataBefore.getHiddenStandardDirectionOrders();
+
+        if (!Objects.equals(hiddenSDOs, previousHiddenSDOs)) {
+            return hiddenSDOs.stream()
+                .filter(removedSDO -> !previousHiddenSDOs.contains(removedSDO))
+                .findFirst()
+                .map(Element::getValue);
+        }
+        return Optional.empty();
+    }
+
+    private Optional<CaseManagementOrder> getRemovedCMO(CaseData caseData, CaseData caseDataBefore) {
+        List<Element<CaseManagementOrder>> hiddenCMOs = caseData.getHiddenCMOs();
+        List<Element<CaseManagementOrder>> previousHiddenCMOs = caseDataBefore.getHiddenCMOs();
+
+        if (!Objects.equals(hiddenCMOs, previousHiddenCMOs)) {
+            return hiddenCMOs.stream()
+                .filter(removedCMO -> !previousHiddenCMOs.contains(removedCMO))
+                .findFirst()
+                .map(Element::getValue);
+        }
+        return Optional.empty();
```

Comment:
Both methods are very similar also, perhaps a good place to use generics and pass in `List<Element<RemovableOrder>>` as params.

Reply:
I tried using generics but `getRemovalReason` is not available in the `RemovalOrder` so, I have to keep 2 methods